### DISCUSSION TITLE
Is PlanetScale database supported as cache storage ?

### DISCUSSION THREAD

**Discussion by LilaRest** [zilliztech/GPTCache#468] [General]
Hi! Thanks for this amazing project.
I wonder if PlanetScale (which is MySQL-based) is supported as a cache storage DB ?

**Comment by SimFG** [comment]
Hi, @LilaRest.
After briefly looking into it, I'm unsure if it's accurate, but it appears that PlanetScale is compatible with MySQL. You can find more information about this here: https://planetscale.com/docs/tutorials/connect-mysql-gui.
If this is the case, you should be able to use PlanetScale as a replacement for MySQL, meaning you can connect to it directly using a URL similar to "mysql+pymysql://root:123456@127.0.0.1:3306/mysql".

**Comment by LilaRest** [comment]
PlanetScale is not fully compatible with MySQL ([see this article](https://planetscale.com/docs/reference/mysql-compatibility)) so I was wondering if anyone already tried that in practice.
I should give it a try in next months :)

**Reply by SimFG** [reply]
I haven't try it. When you try it and get a result, welcome to share the result

**Reply by LilaRest** [reply]
Sure I will!
By the, we are considering the incorporation of GPTCache into a project. However, I'm not enough familiar with its functionning to approximately gauge on costs and performance for our specific use case. Would you be open to a quick DM about this?

**Reply by SimFG** [reply]
It is not very sure how much performance and cost will be improved, because different real scenarios are different. This mainly depends on the probability of similar problems in the scenario where GPTCache is applied. At the same time, if the data accumulated in the cache gradually increases, the hit rate of the cache should gradually increase. In our real product, there is about a 20% probability

**Reply by LilaRest** [reply]
Does it cache per-prompt results or per-meaning results ? I mean it's very unlikely that two users would perform the exact same prompt.
EDIT: I'm reading the doc ^^

**Reply by LilaRest** [reply]
Okay so it uses an embedding model to extract sense of each prompt. So it may not fit in our case as we already have a dedup layer based on OpenAi Embeddings. Thanks for your time @SimFG

**Reply by SimFG** [reply]
yes, it can. And it only needs to be similar, not exactly the same. However, as for how to extract the user's similarity to determine the user's problem, this may have to be dealt with in the real application.

For example, if the user's question is only a small part of the string template, then the best way to determine whether the user is similar should be to extract the user's input and remove the template content.
